mediapicker/src/main/java/org/wordpress/android/mediapicker/viewmodel/MediaPickerViewModel.kt

```diff
@@ -514,12 +525,20 @@ internal class MediaPickerViewModel @Inject constructor(
             is ChooseFromAndroidDevice -> {
                 getSystemPickerAction(action.allowedTypes, canMultiselect)
             }
+
             is CapturePhoto -> {
                 capturedPhotoPath = mediaPickerUtils.generateCapturedImagePath()
                 OpenCameraForPhotos(capturedPhotoPath)
             }
+
             is SwitchSource -> {
-                SwitchMediaPicker(mediaPickerSetupFactory.build(action.dataSource, canMultiselect))
+                SwitchMediaPicker(
+                    mediaPickerSetupFactory.build(
+                        action.dataSource,
+                        MediaTypes.fromAllowedTypes(mediaPickerSetup.allowedTypes),
```

Comment:
Instead of using the media types defined in the Factory before, we now use the types defined in the current picker for the new one, the practical example of this is:
1. We open the device media picker with document types listed as well.
2. Even though it can't list them, when the user click on the system picker button, it'll show the document files.